daft/dataframe/dataframe.py

```diff
@@ -280,13 +280,61 @@ def from_pydict(cls, data: Dict[str, InputListType]) -> "DataFrame":
             )
 
         data_vpartition = Table.from_pydict(data)
-        result_pset = LocalPartitionSet({0: data_vpartition})
+        return cls._from_table(data_vpartition)
+
+    @classmethod
+    @DataframePublicAPI
+    def from_arrow(cls, data: "pa.Table") -> "DataFrame":
```

Comment:
should we extend these to take in `Union[pa.Table, List[pa.Table]]` so we can create a partitioned dataframe